src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -102,11 +102,6 @@
 
         gb.withRateLimitHandler(CUSTOMIZED);
         OkHttpClient client = new OkHttpClient().setProxy(getProxy(defaultIfBlank(apiUrl, GITHUB_URL)));
-        client.setCache(GitHubClientCacheOps.toCacheDir().apply(config));
-        if (config.getClientCacheSize() > 0) {
-            Cache cache = toCacheDir().apply(config);
-            client.setCache(cache);
-        }
```

Comment:
> it is difficult imagining the scenario: Github + Multibranch Pipeline + OrganizationFolder - webhooks

Why? You are running Jenkins inside a private network—normal—and using github.com with private repositories—normal—and are not inclined to set up a reverse proxy for webhook delivery (or do not have permission from IT)—normal. So you need to fall back to polling every fifteen minutes or whatever.